pkg/controller/ensure_deployment_test.go

```diff
@@ -62,7 +62,7 @@ func TestEnsureDeploymentHandler(t *testing.T) {
 			migrationHash: "testtesttesttest",
 			secretHash:    "secret",
 			existingDeployments: []*appsv1.Deployment{{ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{
-				metadata.SpiceDBConfigKey: "n549h6bh675h699h697h86h5cchdfq",
+				metadata.SpiceDBConfigKey: "n59hb4h5f7h5b7h5dchddh79h67dq",
```

Comment:
Does these keys changing imply that existing operators, when updated, will cause all resources to be re-applied?

Reply:
It does, I didn't look into why this happened specifically, but we did jump from 1.26 client go to 1.28 so I assume it's something in the deployment fields.